pkg/kds/mux/session.go

```diff
@@ -126,39 +171,30 @@ func (s *session) Error() <-chan error {
 	return s.err
 }
 
-type sendItem struct {
-	msg     *mesh_proto.Message
-	errChan chan error
-}
-
 type bufferStream struct {
-	sendBuffer chan sendItem
+	sendBuffer chan *mesh_proto.Message
 	recvBuffer chan *mesh_proto.Message
 
 	// Protects the send-buffer against writing on a closed channel, this is needed as we don't control in which goroutine `Send` will be called.
 	lock   sync.Mutex
 	closed bool
 }
 
-func newBufferStream() *bufferStream {
+func newBufferStream(bufferSize uint32) *bufferStream {
 	return &bufferStream{
-		sendBuffer: make(chan sendItem, 1),
-		recvBuffer: make(chan *mesh_proto.Message, 1),
+		sendBuffer: make(chan *mesh_proto.Message, bufferSize),
+		recvBuffer: make(chan *mesh_proto.Message, bufferSize),
 	}
 }
 
 func (k *bufferStream) Send(message *mesh_proto.Message) error {
 	k.lock.Lock()
+	defer k.lock.Unlock()
 	if k.closed {
-		k.lock.Unlock()
 		return io.EOF
 	}
-	errChan := make(chan error)
-	k.sendBuffer <- sendItem{msg: message, errChan: errChan}
-
-	k.lock.Unlock()
-	r := <-errChan
-	return r
+	k.sendBuffer <- message
+	return nil
```

Comment:
It will be ok.

If send fails and we don't immediately return an error, it's fine because we are doing "recv" shortly after anyways and failed stream will error the recv.